### PR TITLE
Functions and Trigger commands

### PR DESCRIPTION
**Pull request by JadeRedworth**
Updated apps to use appID instead of name.
Added functions and triggers command. 
Invoking a function/trigger is not yet supported.
Added tests around the crud operations for functions and triggers.

Note: In order to handle backward compatibility issue, we are using the v1 client in apps commands for now.

_Both fn and fn_go have the trigger branches in master_.

### CODE REVIEW

**Comment by JadeRedworth** [comment]
I think we should. The client is initialized on each app command, is there a way to check what API server is being used so we can create the client dependent on that?

Also are we planning to hide the functions/triggers commands initally? @owencliffe @msgodf

**Comment by zootalures** [comment]
Not sure using the api-url works in general here - currently it's not canonically with or without the version suffix (the provider currently normalises it based on this - that's how it provides both v1 and v2 clients.

**Comment by rikgibson** [comment]
Discussed this more with @zootalures - checking the api-url won't quite work. Wondering if we have the CLI enforce a minimum required server version somehow? Checking the Server version does require an API call, which we don't really want to have to make on each CLI invocation, but we could cache the last-seen version for a given context in the context .yaml, and only look up if that is not present or lower than the version required by the CLI? If the resultant version is still too low, simply message that the server version is too low and to upgrade the server - might only need something like this when using a default context.

**Comment by zootalures** [comment]
Maybe just on error?

File: objects/fn/commands.go [hunk 1]
```diff
@@ -0,0 +1,221 @@
+package fn
```
[the remark this answers is not in the document]
Comment (mantree) [reply objects/fn/commands.go]: 'fn' is correct imo

[the remark this answers is not in the document]
Comment (JadeRedworth) [reply objects/fn/commands.go]: @mantree we are battling between using `fn` in the as a short name for functions:
`fn create fn ...` seem's slightly weird?

**Pull request merged**